File: Programmers/Inho/Level01/문자열 내 마음대로 정렬하기.cs
using System;
public class Solution {
    public string[] solution(string[] strings, int n) {
        string[] answer = new string[] {"a"};
        string temp;
        for(int i = 0; i<strings.Length;i++){
            for(int j=i;j<strings.Length;j++){
                if(strings[i][n] == strings[j][n])
                {
                    if(Compare(strings[i], strings[j])){
                        temp = strings[i];
                        strings[i] = strings[j];
                        strings[j] = temp;
                    }
                }
                else if(strings[i][n]>strings[j][n])
                {
                    temp = strings[i];
                    strings[i] = strings[j];
                    strings[j] = temp;
                }
            }
        }
        return strings;
    }
    public bool Compare(string str1, string str2){
        int max = Math.Max(str1.Length, str2.Length);
        for(int i =0 ;i<max;i++){
            if(str1[i]==str2[i])
                continue;
            if(str1[i]>str2[i]){
                return true;
            }else
                return false;
        }
        return true;
    }
}

File: Programmers/Inho/C++/Level01/문자열 내 마음대로 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(vector<string> strings, int n) {
    vector<string> answer;
    vector<string> temp;
    for(int i = 0;i<26;i++){
        temp.clear();
        for(string st:strings){
            if(st[n]==i+'a'){
                temp.push_back(st);
            }
        }
        sort(temp.begin(), temp.end());
        for(string st:temp){
            answer.push_back(st);
        }
    }
    return answer;
}